app_dart/lib/src/request_handlers/scheduler/vacuum_stale_tasks.dart

```diff
@@ -53,32 +53,16 @@ class VacuumStaleTasks extends RequestHandler<Body> {
     final DatastoreService datastore = datastoreProvider(config.db);
 
     final List<FullTask> tasks = await datastore.queryRecentTasks(slug: slug).toList();
-    final Set<Task> tasksToBeReset = <Task>{};
+    final List<Task> tasksToBeReset = <Task>[];
     for (FullTask fullTask in tasks) {
       final Task task = fullTask.task;
-      if (task.status != Task.statusInProgress) {
-        continue;
-      }
-
-      if (task.createTimestamp == null) {
-        log.fine('Vacuuming $task due to createTimestamp being null');
+      if (task.status == Task.statusInProgress && task.buildNumber == null) {
```

Comment:
Do we need to check how long the task has been in progress before resetting?

Reply:
The old logic to be based on `createTimestamp` caused duplicated scheduling, and that's why I switched to use `buildNumber==null`. Considering the running frequency is super low, I would not worry too much about the timestamp check.